### PR TITLE
Add Shell Sort 🐙

### PR DESCRIPTION
**Pull request by NghiaTranUIT**
## What's it?
This PR introduces ShellSort visualization. 

The implementation is inspired from [Shell Sort - Python - Geeksforgeeks](https://www.geeksforgeeks.org/shellsort/)
 
## Changelogs
- [x] ShellSort logic
- [x] New sort item in the dropdown selection

## Why?
- Visualization is fun 🌮 
- Swag from Hacktoberfest 🧧

### CODE REVIEW

File: algorithms.js [hunk 1]
```diff
@@ -215,3 +215,21 @@ function isSorted(values) {
   }
   return true;
 }
+
+function* shellSort(values) {
+  let n = Math.floor(values.length)
```
Comment (khoi) [inline algorithms.js]: `n` doesn't need to be floored ;D

File: algorithms.js [hunk 2]
```diff
@@ -215,3 +215,21 @@ function isSorted(values) {
...
+      let tmp = values[i];
+      let j = i;
+      while (j >= gap && values[j - gap] > tmp) {
+        values[j] = values[j - gap];
+        j -= gap ;
+      }
+      values[j] = tmp;
+      yield;
+    }
+    gap = Math.floor(gap / 2);
```
Comment (khoi) [inline algorithms.js]: Can we use `for` style loop here so it matches the textbook implementation on Wikipedia or GeeksforGeeks

**Review by khoi — CHANGES_REQUESTED** [review]
Awesome, thanks for contributing

**Comment by NghiaTranUIT** [comment]
Thank for pointing it out 🌮 Please do a second review 🙌

File: algorithms.js [hunk 3]
```diff
@@ -215,3 +215,18 @@ function isSorted(values) {
...
 }
+
+function* shellSort(values) {
+  let n = values.length;
+  for (let gap = Math.floor(n / 2); gap > 0; gap = Math.floor(gap / 2)) {
+    for (let i = gap; i < n; i++) {
+      let tmp = values[i];
+      let j = i;
+      for (j = i; j >= gap && values[j - gap] > tmp; j -= gap) {
+        values[j] = values[j - gap];
```
Comment (khoi) [inline algorithms.js:226]: Missing `yield` here possibly?

**Review by khoi — CHANGES_REQUESTED** [review]

File: algorithms.js (hunk 3 above)
Comment (NghiaTranUIT) [reply algorithms.js:226]: opp, I'm noob JS 😮

**Comment by khoi** [comment]
#2

**Comment by khoi** [comment]
@all-contributors please add @NghiaTranUIT for his code contribution

**Comment by khoi** [comment]
@all-contributors please add @NghiaTranUIT for his code contribution

**Pull request merged**